R/object.R

```diff
@@ -29,6 +29,11 @@ new_object <- function(.data = NULL, ...) {
     )
   }
 
+  missing_props <- nms[vlapply(args, is.null)]
+  for(prop in missing_props) {
+    args[[prop]] <- prop_default(obj_cls@properties[[prop]])
+  }
```

Comment:
Does this limit you from setting a property to `NULL` at all? Since now `NULL` also means "go get me the default"?

Reply:
Oh yeah, good catch. I've switched sentinel value to `missing_class`, which while isn't exactly the same meaning seems close enough, and avoids the challenges of working with `missing()`.